src/main/java/org/embulk/input/marketo/MarketoUtils.java

```diff
@@ -0,0 +1,175 @@
+package org.embulk.input.marketo;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.base.Function;
+import com.google.common.base.Strings;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableList;
+import org.embulk.base.restclient.ServiceResponseMapper;
+import org.embulk.base.restclient.jackson.JacksonServiceRecord;
+import org.embulk.base.restclient.jackson.JacksonServiceResponseMapper;
+import org.embulk.base.restclient.jackson.JacksonTopLevelValueLocator;
+import org.embulk.base.restclient.record.ServiceRecord;
+import org.embulk.base.restclient.record.ValueLocator;
+import org.embulk.spi.Column;
+import org.embulk.spi.ColumnVisitor;
+import org.embulk.spi.DataException;
+import org.embulk.spi.Exec;
+import org.embulk.spi.Schema;
+import org.embulk.spi.type.Types;
+import org.slf4j.Logger;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by tai.khuu on 9/18/17.
+ */
+public class MarketoUtils
+{
+    private static final Logger LOGGER = Exec.getLogger(MarketoUtils.class);
+    public static final String ISO_8601_FORMAT = "%Y-%m-%dT%H:%M:%S%z";
+
+    public static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    public static Function<ObjectNode, ServiceRecord> transformObjectToJackSonServiceRecordFunction()
+    {
+        return new Function<ObjectNode, ServiceRecord>()
+        {
+            @Nullable
+            @Override
+            public JacksonServiceRecord apply(@Nullable ObjectNode input)
+            {
+                return new JacksonServiceRecord(input);
+            }
+        };
+    }
+
+    private MarketoUtils()
+    {
+    }
+
+    public static ServiceResponseMapper<? extends ValueLocator> buildDynamicResponseMapper(List<Column> columns)
+    {
+        JacksonServiceResponseMapper.Builder builder = JacksonServiceResponseMapper.builder();
+        for (Column column : columns) {
+            if (column.getType().equals(Types.TIMESTAMP)) {
+                builder.add(new JacksonTopLevelValueLocator(column.getName()), column.getName(), column.getType(), ISO_8601_FORMAT);
+            }
+            else {
+                builder.add(new JacksonTopLevelValueLocator(column.getName()), column.getName(), column.getType());
+            }
+        }
+        return builder.build();
+    }
+
+    public static List<String> getFieldNameFromSchema(Schema schema)
+    {
+        ImmutableList<String> fieldNames = FluentIterable.from(schema.getColumns()).transform(new Function<Column, String>()
```

Comment:
Can return the list directly without temp var.

Reply:
Fixed